subsys/net/ip/tcp.c

```diff
@@ -1932,13 +1944,15 @@ static void tcp_queue_recv_data(struct tcp *conn, struct net_pkt *pkt,
 			}
 		}
 
-		if (IS_ENABLED(CONFIG_NET_TCP_LOG_LEVEL_DBG)) {
-			if (inserted) {
-				NET_DBG("All pending data: conn %p", conn);
-				print_seq_list(conn->queue_recv_data->buffer);
-			} else {
-				NET_DBG("Cannot add new data to queue");
+		if (inserted) {
+			NET_DBG("All pending data: conn %p", conn);
+			if (check_seq_list(conn->queue_recv_data->buffer) == false) {
+				/* error in sequence list, drop it */
+				net_buf_unref(conn->queue_recv_data->buffer);
```

Comment:
So here we could print something like this
```
NET_ERR("Incorrect order in out of order sequence, conn %p", conn);
```

Reply:
Indead